app.py

```diff
@@ -721,21 +721,39 @@ def jurisdiction_retrieval_list(election_id, jurisdiction_id, round_num):
     election = get_election(election_id)
     csv_io = io.StringIO()
     retrieval_list_writer = csv.writer(csv_io)
-    retrieval_list_writer.writerow(["Batch Name","Ballot Number","Storage Location","Tabulator","Ticket Number","Audit Board"])
+    retrieval_list_writer.writerow(["Batch Name","Ballot Number","Storage Location","Tabulator","Ticket Numbers","Audit Board"])
 
     # check the jurisdiction and round
     jurisdiction = Jurisdiction.query.filter_by(election_id = election.id, id = jurisdiction_id).one()
     round = Round.query.filter_by(election_id = election.id, round_num = round_num).one()
 
+    '''
     ballots = SampledBallotDraw.query.filter_by(round_id = round.id) \
                     .join(SampledBallotDraw.batch).filter_by(jurisdiction_id = jurisdiction_id)  \
                     .join(SampledBallotDraw.sampled_ballot).join(SampledBallot.audit_board) \
                     .add_entity(Batch).add_entity(AuditBoard) \
+                    .group_by(Batch.name)\
+                    .group_by(SampledBallotDraw.ballot_position) \
                     .order_by(AuditBoard.name, Batch.name, SampledBallotDraw.ballot_position, SampledBallotDraw.ticket_number) \
                     .all()
+    '''
 
-    for ballot_draw, batch, audit_board in ballots:
-        retrieval_list_writer.writerow([batch.name, ballot_draw.ballot_position, batch.storage_location, batch.tabulator, ballot_draw.ticket_number, audit_board.name])
+    # Get deduped sampled ballots
+    ballots = SampledBallotDraw.query.filter_by(round_id = round.id) \
+                    .join(SampledBallotDraw.batch).filter_by(jurisdiction_id = jurisdiction_id)  \
+                    .join(SampledBallotDraw.sampled_ballot).join(SampledBallot.audit_board) \
+                    .add_entity(Batch).add_entity(AuditBoard) \
+                    .group_by(Batch.name, Batch.id, Batch.storage_location, Batch.tabulator, AuditBoard.name)\
+                    .group_by(SampledBallotDraw.ballot_position) \
+                    .values(Batch.id, SampledBallotDraw.ballot_position, Batch.name, Batch.storage_location, Batch.tabulator, AuditBoard.name)
+
+    rows = []
+    for batch_id, position, batch_name, storage_location, tabulator, audit_board in ballots:
+
+        # Get ticket_numbers
+        ticket_numbers = ','.join([b.ticket_number for b in SampledBallotDraw.query.filter_by(batch_id = batch_id, ballot_position=position, round_id = round.id).all()])
```

Comment:
ah, so this will issue another query within the loop, which means now we're hammering the database a lot. I don't recommend this, it will come back to bite us the moment we have any significant load.

What were you going for with the change to the outer query btw?